### PR TITLE
feat(webidl2): Update to v23.11

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
	- <https://github.com/w3c/webidl2.js/blob/gh-pages/CHANGELOG.md#v23100-2019-10-14>
	- <https://github.com/w3c/webidl2.js/compare/v23.8.3...v23.11.0>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/webidl2/index.d.ts [hunk 1]
```diff
@@ -1,40 +1,90 @@
...
+    sourceName: string | undefined;
+
     /** a short peek at the text at the point where the error happened */
     input: string;
     /** the five tokens at the point of error, as understood by the tokeniser */
     tokens: ValueDescription[];
+}
 
-    toString(): string;
+export interface SimpleTypeDescription {
```
Comment (saschanaz) [inline types/webidl2/index.d.ts:67]: How about SingleType, as generic types are not "simple"?

File: types/webidl2/index.d.ts [hunk 2]
```diff
@@ -257,6 +321,41 @@ export interface ExtendedAttributeRightHandSideIdentifierList {
...
+    value: string;
+}
+
+export interface ExtendedAttributeRightHandSideIntegerList {
+    type: "integer-list";
+    value: ExtendedAttributeRightHandSideInteger[];
+}
+
+export interface Token {
+    type: "float" | "integer" | "identifier" | "string" | "whitespace" | "other";
```
Comment (saschanaz) [inline types/webidl2/index.d.ts:355]:
```suggestion
    type: "decimal" | "integer" | "identifier" | "string" | "whitespace" | "other";
```

**Review by saschanaz — CHANGES_REQUESTED** [review]
Generally LGTM with some requests:

**Review by saschanaz — APPROVED** [review]
Thanks! 😁

**Comment by ExE-Boss** [comment]
Looks like the bot didn’t correctly remove the <kbd>[Revision needed](https://github.com/DefinitelyTyped/DefinitelyTyped/labels/Revision%20needed)</kbd> label.

---

/cc @RyanCavanaugh

**Pull request merged**